### PR TITLE
REF: Refactor merge array conversions for factorizer

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -2433,6 +2405,41 @@ def _factorize_keys(
     return llab, rlab, count
 
 
+def _convert_arrays_and_get_rizer_klass(
```
Comment (jbrockmendel) [inline pandas/core/reshape/merge.py:2408]: pure cut/paste?

Comment (phofl) [reply pandas/core/reshape/merge.py:2408]: Yep, agreed to this in the initial pr

**Review by mroeschke — APPROVED** [review]
LGTM merge when ready @jbrockmendel

**Comment by jbrockmendel** [comment]
thanks @phofl

**Pull request merged**